src/series/bar.js

```diff
@@ -34,14 +34,15 @@
                         barBottom = yScale(y0),
                         barTop = yScale(y0 + y1Value(d, i));
 
-                    var g = d3.select(this)
-                        .attr('transform', 'translate(' + x + ', ' + barTop + ')');
-
                     pathGenerator.x(d3.functor(0))
                         .y(d3.functor(0))
                         .height(function() { return barBottom - barTop; });
 
-                    g.select('path')
+                    var barGroup = d3.select(this);
+
+                    d3.transition(barGroup)
+                        .attr('transform', 'translate(' + x + ', ' + barTop + ')')
```

Comment:
I think the enter transitions would look a bit nicer if this translation was also applied to the [enter selection](https://github.com/ScottLogic/d3-financial-components/pull/376/files#diff-8eef6e441ba0cee7bd1c8bde8fbba961L22), but it's not a big deal for now and something that can be changed later. 

Although, it seems this would be a bit harder to do now that the convenience accessors are encapsulated within the `g.each`?